django/db/models/expressions.py

```diff
@@ -1144,23 +1150,6 @@ def as_sql(self, compiler, connection, template=None, **extra_context):
         params *= template.count('%(expression)s')
         return (template % placeholders).rstrip(), params
 
-    def as_sqlite(self, compiler, connection):
-        template = None
-        if connection.Database.sqlite_version_info < (3, 30, 0):
-            if self.nulls_last:
-                template = '%(expression)s IS NULL, %(expression)s %(ordering)s'
-            elif self.nulls_first:
-                template = '%(expression)s IS NOT NULL, %(expression)s %(ordering)s'
-        return self.as_sql(compiler, connection, template=template)
-
-    def as_mysql(self, compiler, connection):
-        template = None
-        if self.nulls_last:
-            template = 'ISNULL(%(expression)s), %(expression)s %(ordering)s '
-        elif self.nulls_first:
-            template = 'IF(ISNULL(%(expression)s),0,1), %(expression)s %(ordering)s '
-        return self.as_sql(compiler, connection, template=template)
```

Comment:
Note: The official [MariaDB documentation](https://mariadb.com/kb/en/library/null-values/#ordering) recommends to use `ISNULL()`, but it seems that it's exactly the same.

Reply:
True, although ironically they then do `IF(col1 IS NULL, 0, 1)` in the second example instead of `col1 IS NOT NULL`... :man_facepalming:

Reply:
Seems fine with expressions too: [fiddle](https://dbfiddle.uk/?rdbms=mariadb_10.4&fiddle=27dfbd5fbbc411aeeb58d7ddbd9007ef).